File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int ans = 0;
        int i = 0;
        int j = 0;
        int mid = 0;
        for (int index = 0; index < nums2.size(); index++){
            i = 0;
            j = std::min(index, nums1.size() - 1)
            while (i <= j){
                mid = (i + j) / 2;
                if (nums1[mid] > nums2[index]){
                    i = mid + 1;
                }
                else{
                    ans = std::max(ans, index - mid);
                    j = mid - 1;
                }
            }
        }
        return ans;
    }
};

File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        ans = 0
        for index in range(len(nums2)):
            i = 0
            j = min(index, len(nums1) - 1)

            # We are looking for first num in nums1 such a num is the
            # smallest but satisfies num > nums2[r]
            while i <= j:
                mid = (i + j) // 2
                if nums1[mid] > nums2[index]:
                    i = mid + 1
                else:
                    ans = max(ans, index - mid)
                    j = mid - 1
        return ans
